rlberry/utils/check_agent.py

```diff
@@ -0,0 +1,202 @@
+    set_external_seed(SEED)
+    actions1 = []
+    for f in range(5):
+        # test on 5 actions
+        actions1.append(agent1.agent_handlers[0].policy(state))
+    set_external_seed(SEED)
+    for f in range(5):
+        # test on 5 actions
+        action2 = agent2.agent_handlers[0].policy(state)
+        if np.mean(actions1[f] == action2) != 1:
+            result = False
+    return result
+
+
+def check_fit_additive(Agent, continuous_state=False):
+    """
+    check that fitting two times with 10 fit budget is the same as fitting
+    one time with 20 fit budget.
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    env_kwargs = {}
+    agent1 = AgentManager(
+        Agent, (env_ctor, env_kwargs), fit_budget=5, n_fit=1, seed=SEED
+    )
+    agent1.fit(10)
+    agent1.fit(10)
+
+    agent2 = AgentManager(
+        Agent, (env_ctor, env_kwargs), fit_budget=5, n_fit=1, seed=SEED
+    )
+    agent2.fit(20)
+
+    env = env_ctor()
+    state = env.reset()
+    result = check_agent_almost_equal(state, agent1, agent2)
+
+    assert (
+        result
+    ), "Error: fitting the agent two times for 10 steps is not equivalent to fitting it one time for 20 steps."
+
+
+def check_save(Agent, continuous_state=False):
+    """
+    Check that the agent save a non-empty file.
+
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        if continuous_state:
+            env_ctor = PBall2D
+        else:
+            env_ctor = Chain
+        env_kwargs = {}
+        agent1 = AgentManager(
+            Agent,
+            (env_ctor, env_kwargs),
+            fit_budget=5,
+            n_fit=1,
+            seed=SEED,
+            output_dir=tmpdirname,
+        )
+        agent1.fit(10)
+        assert (
+            os.path.getsize(str(agent1.output_dir_) + "/agent_handlers/idx_0.pickle")
+            > 1
+        ), "the file saved appear to be empty"
+
+
+def check_seeding_agent(Agent, continuous_state=False):
+    """
+    Check that the agent is reproducible.
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    agent1 = check_agent_manager(Agent, continuous_state)
+    agent2 = check_agent_manager(Agent, continuous_state)
```

Comment:
because it is faster than rewriting the fit and check_agent_manager already fit so I figured, let us use it to fit an agent.

Reply:
I think I was surprised that a `check_`... function returns something because usually, they return nothing or a bool. However, I am having a hard time coming up with an alternative - maybe adding a `safe_manager_wrap` function that tries to wrap an agent (similarly to `safe_reseed`). Then. we could use it for the checks and also expose it so people can see.

Reply:
Ok, good idea, I will use _fit_agent_manager that returns a fitted agent and check_agent_manager that just do the check. You are right this is easier to understand.